### ISSUE TITLE
[Scarpet suggestion] Corner-to-corner and corner-to-directional-offest versions of rect()

### ISSUE THREAD

**Issue by TheCatSaber** [gnembon/fabric-carpet#492]
Currently `rect()` creates an iterator over corner-to-offest-in-all-directions.
Corner-to-corner would create an iterator over between the two corners (like vanilla fill command, just an iterator).
Corner-to-directional-offest would create an iterator from one corner to corner+directional-offest for each of the dimensions.

These would reduce head hurting when trying to test over an area where you know the corners. (See my problem [here on SciCraft Discord](https://discordapp.com/channels/211786369951989762/573613501164159016/762974963111690250))

**Comment by Ghoulboy78** [comment]
And just as the good people in the carpet mod chat, I would advise you to look into the `volume()` function, @TheCatSaber

**Comment by Ghoulboy78** [comment]
https://github.com/gnembon/fabric-carpet/blob/master/docs/scarpet/Full.md#volumefrom_pos-to_pos-expr

**Comment by gnembon** [comment]
yes, `volume` is a loop that does corner to corner and you don't need to worry which corners you specify. `scan' is a loop center oriented/offset.
`rect` is like `scan` but an iterator. There is no iterator for `volume`, but you can have a function that converts it and return it since iterators can be passed along by value, unlike loops.

I get that - the naming (volume, scan, rect) is kinda random. Didn't come up with better names back in the days.
